docs/spec.md

```diff
@@ -2680,6 +2679,66 @@ op, but the result shape is specified dynamically via `output_shape`.
 
 &nbsp;[More Examples](https://github.com/openxla/stablehlo/tree/main/stablehlo/tests/interpret/dynamic_iota.mlir)
 
+### dynamic_pad
+
+#### Semantics
+
+This operation is functionally identical to
+[pad](https://github.com/openxla/stablehlo/blob/main/docs/spec.md#pad)
+op, but the amounts of padding are specified dynamically via `edge_padding_low`,
+`edge_padding_high` and `interior_padding`.
```

Comment:
```
but with `edge_padding_low`, `edge_padding_high` and `interior_padding` specified dynamically as operands.
```

Let's use a consistent wording for all dynamic ops, and I like the notion of:

This operation is semantically equivalent to `<OP>`, with `<attribute_names>` specified dynamically as operands.